src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,78 @@
 import React from 'react';
+import { todoListType } from '../../types';
+import { Todo } from '../Todo';
 import './TodoList.scss';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends React.PureComponent {
+  state = {
+    search: '',
+    filteredTodos: 'all',
+  }
+
+  filter = {
+    all: () => true,
+    completed: todo => todo.completed,
+    active: todo => !todo.completed,
+  }
+
+  changeHandler = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
+
+  render() {
+    const { todos, selectedUserId, onButtonClick } = this.props;
+    const { search, filteredTodos } = this.state;
+    const todoIncludes = todo => todo.title
+      .toLowerCase()
+      .includes(search.toLowerCase());
+
+    const visibleTodos = todos
+      .filter(todo => todo.title
+        && todoIncludes(todo)
+          && this.filter[filteredTodos](todo));
```

Comment:
let's create separate functions filterByTitle and filterByStatus so we can use them like this

```suggestion
    const visibleTodos = todos
      .filter(this.filterByTitle)
      .filter(this.filterByStatus);
```

hint: each of them would automatically accept ONE todo as argument when passed as callback to filter